### ISSUE TITLE
Document MM_ENV

### ISSUE THREAD

**Issue by jobevers** [middleman/middleman#2271]
Please document `MM_ENV`.

A wonderful place for it would be in this section of the documentation: https://middlemanapp.com/advanced/configuration/#environment-specific-settings

**Comment by tdreyno** [comment]
Can you describe your specific use case?

**Comment by jobevers** [comment]
Having used middleman a bit more over the past few days its now clear to me that `middleman server` is "development" and `middleman build` is "build" but at the time it was not obvious to me how the  the development section of the config file  would ever be reached.  

I had been exclusively using `middleman build` and I looked at the help of `middleman build` and there was nothing in there about switching environments. I eventually discovered `MM_ENV` and it seemed like that should be documented.

Documenting MM_ENV doesn't seem necessary anymore, but I think it would be useful to add a sentence to that section to reiterate that `configure :development...` gets  run by default when `middleman server` is used and `configure :build...` gets run by default when `middleman build` is used.
